src/Components/PropertyInspector/PropertyInspectoryModel.ts

```diff
@@ -344,21 +384,116 @@ abstract class PropertyInspectorModel {
                                                   .mapValue,
                                               path: PropertyInspectorModel.buildPath(
                                                   path,
-                                                  modelProperty.name
+                                                  modelProperty.name,
+                                                  modelProperty.index
                                               ),
                                               propertySourceObject: mapValue,
                                               mapInfo: { key },
                                               isMapChild: true,
                                               forceSet: true,
-                                              schemas
+                                              schemas,
+                                              isArrayItem
                                           }
                                       );
                                   })
                                 : null,
                         ...(schemas && { mapSchemas: schemas })
                     };
                 }
-                case DTDLSchemaType.Array: // TODO support arrays in future
+                case DTDLSchemaType.Array: {
+                    const arrayItems: any[] = PropertyInspectorModel.getPropertyValueOrDefault(
+                        modelProperty.name,
+                        propertySourceObject,
+                        dtdlPropertyTypesEnum.Array
+                    );
+                    const targetSchema = schemas?.find(
+                        (s) =>
+                            s['@id'] ===
+                            ((modelProperty.schema as any).elementSchema ||
+                                modelProperty.schema)
+                    );
+                    return {
+                        name: modelProperty.name,
+                        displayName: PropertyInspectorModel.parsePropertyTreeDisplayName(
+                            {
+                                name: modelProperty.name,
+                                ...modelProperty
+                            }
+                        ),
+                        role: NodeRole.parent,
+                        schema: dtdlPropertyTypesEnum.Array,
+                        isCollapsed: true,
+                        type: DTDLType.Property,
+                        path: mapInfo
```

Comment:
what is `mapInfo`? I kind of assumed when I saw it up above that it was tied to map objects, but now we are inside of knowing the item is an array so I'm questioning that assumption.

Reply:
Good point. This'll always be false.